### PR TITLE
[WIP] Api example: Geojson integration

### PR DESCRIPTION
**Pull request by kandre**
[Testlink](http://mf-chsdi3.dev.bgdi.ch/kan_geojson_example/examples/geoadmin_geojson.html)

This PR should be merged as soon modifications in our API are merged (and hence this code adapted accordingly and squashed).

### CODE REVIEW

**Comment by cedricmoullet** [comment]
There is no tooltip ?

2015-05-13 17:25 GMT+02:00 kandre <notifications@github.com>:

> Testlink
> <http://mf-chsdi3.dev.bgdi.ch/kan_geojson_example/examples/geoadmin_geojson.html>
>
> This PR should be merged as soon modifications in our API are merged (and
> hence this code adapted accordingly and squashed).
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/geoadmin/mf-chsdi3/pull/1444
> Commit Summary
>
>    - Start
>    - Start
>    - Workable example
>    - Cleaned
>    - Add button onclick
>    - Alert messages
>    - Set Panel
>    - Set panel 2
>
> File Changes
>
>    - *A* chsdi/static/doc/examples/geoadmin_geojson.html
>    <https://github.com/geoadmin/mf-chsdi3/pull/1444/files#diff-0> (55)
>    - *A* chsdi/static/doc/examples/geoadmin_geojson.js
>    <https://github.com/geoadmin/mf-chsdi3/pull/1444/files#diff-1> (98)
>    - *A* chsdi/static/doc/examples/stylesFromLiterals.js
>    <https://github.com/geoadmin/mf-chsdi3/pull/1444/files#diff-2> (178)
>
> Patch Links:
>
>    - https://github.com/geoadmin/mf-chsdi3/pull/1444.patch
>    - https://github.com/geoadmin/mf-chsdi3/pull/1444.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoadmin/mf-chsdi3/pull/1444>.
>



-- 
Twitter: http://twitter.com/cedricmoullet
Linked In: http://www.linkedin.com/in/cedricmoullet

<http://twitter.com/cedricmoullet>

**Comment by loicgasser** [comment]
Need some adaptations in the tooltip, it should not be too hard, I can help if needed. Use the field description and you can do the same than what we have done with KML (same field).
Nice example. would be nice to have a text box containing the remote json style. We could then edit it and resubmit it to change the style. It could be our very first style editor.... ;)

**Comment by gjn** [comment]
This seems to be the last missing item for the geojson release we want to do next week. Please focus on this.

File: chsdi/static/doc/examples/geoadmin_geojson.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  </head>
+  <body>
+    <h1>GeoAdmin API example: integration of geojson layers</h1>
+    <div id="outer" style="overflow: hidden; width: 100%">
+    <div id="managerContainer" class="panel panel-default" style="margin:20px;width:450px;float:left;min-height:350px;">
+    <div class="panel-heading">Geojson loading and styling</div>
+      <div class="panel-body">
+        <div style="padding-bottom:5px">Geojson Url:</div>
+        <form name="geojson-url">
+          <input class="form-control" placeholder="Enter a Geojson file url ..." onClick="this.select();" value="https://mf-chsdi3.dev.bgdi.ch/gal_geojson/ogcproxy?url=http://data.geo.admin.ch/ch.bafu.hydroweb-messstationen_zustand/ch.bafu.hydroweb-messstationen_zustand_en.json" style="width:100%;">
```
Comment (procrastinatio) [inline chsdi/static/doc/examples/geoadmin_geojson.html]: Do not hard code url

File: chsdi/static/doc/examples/geoadmin_geojson.html [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        </form>
+        <div id="geojsonSuccess" class="alert alert-success" style="display:none">Geojson file has been successfully loaded.</div>
+        <div id="geojsonError" class="alert alert-danger" style="display:none">File loading has failed.</div>
+      </div>
+      <div class="panel-body">
+        <div style="padding-bottom:5px">Style Url:</div>
+        <form name="style-url">
+          <input class="form-control" placeholder="Enter a styling json file url ..." onClick="this.select();" value="https://mf-chsdi3.dev.bgdi.ch/gal_geojson/static/vectorStyles/ch.bafu.hydroweb-messstationen_zustand.json" style="width:100%;">
```
Comment (procrastinatio) [inline chsdi/static/doc/examples/geoadmin_geojson.html]: Use relative urls

File: chsdi/static/doc/examples/geoadmin_geojson.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          This example show how to integrate geojson layers and styling them.
+          <a href="geoadmin_geojson.js">See the code</a>
+        </p>
+      </div>
+    </div>
+    <div id="map" class="panel panel-default" style="height:350px;margin:20px;overflow:hidden"></div>
+    </div>
+    <script type="text/javascript" src="/static/js/jquery-2.0.3.min.js"></script>
+    <!-- To be adapted before merge in https://github.com/geoadmin/mf-chsdi3/blob/master/buildout.cfg-->
+    <script src="../uncached_loader.js?mode=debug"  type="text/javascript"></script>
```
Comment (procrastinatio) [inline chsdi/static/doc/examples/geoadmin_geojson.html]: Yuks, _loader.js_ and no _debug_

**Comment by procrastinatio** [comment]
Few comment:
- Explain a bit why your example is awsome. The average user see only colored circle.
- Link example in the examples page
- A basic explanantion on the JSON style file

**Comment by kandre** [comment]
Thanks for your comments. I updated the code with: 
- Tooltip
- GeoJSON styling editor
- More explanations about this example (although I tried too keep it short, in combination with the editor, it may be sufficient)
- Custom GeoJSON and styling files. Those ones allows the user to see he can draw different geometries and styles (but do we prefer to preview real data maybe ?)
- Relative urls

**Comment by gjn** [comment]
Please squash.

File: chsdi/static/doc/examples/geoadmin_geojson.html [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            <a href="geoadmin_geojson.js">See the code</a>
+          </p>
+        <div style="display: none;">
+          <div id="popup" title="KML"></div>
+        </div>
+      </div>
+    <script type="text/javascript" src="/static/js/jquery-2.0.3.min.js"></script>
+    <script src="//netdna.bootstrapcdn.com/bootstrap/3.0.2/js/bootstrap.min.js"></script>
+    <!-- To be adapted before merge in https://github.com/geoadmin/mf-chsdi3/blob/master/buildout.cfg-->
+    <script src="../uncached_loader.js?mode=debug"  type="text/javascript"></script>
```
Comment (kandre) [inline chsdi/static/doc/examples/geoadmin_geojson.html]: will be adapted when API side is ready

**Comment by gjn** [comment]
@kandre Can this be merged for today's deploy, or do you want to update it with https://github.com/geoadmin/ol3/pull/126?

**Comment by kandre** [comment]
Geojson integration #126 in the JS API broke a bit the example. In my opinion it would be wiser to wait for the update.

**Comment by gjn** [comment]
So merge this?

**Comment by loicgasser** [comment]
I believe Styles urls start with localhost because I am using request.static_url in models/bod.py
Can you check that? We should maybe use a buildout variable instead to fix this issue.

**Comment by loicgasser** [comment]
Related to serverconfig stuff

**Comment by gjn** [comment]
Just discussed with @kandre . We'll do another PR with the state of this PR as it was prior to the update yesterday and then merge that for today's deploy.

We'll have the example deployed and can work to improve it in this PR.

**Comment by loicgasser** [comment]
Yeah not related to that pr directly you can merge this one.

**Comment by kandre** [comment]
Yes, as we discussed with the first commit at the moment

**Comment by loicgasser** [comment]
I found a solution that will also improve the loader performance. I'll close this one and reopen a clean PR to get a better understanding of what's going on.

**Pull request closed without merging**